File: dotnet/0.02/WFlow.Core/ActivityStates.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace WFlow
{
    public enum ActivityStates
    { 
        /// <summary>
        /// 创建中，尚未执行过Predicate
        /// </summary>
        Creating,
        /// <summary>
        /// 已经创建，至少执行过一次predicate，但未通过检查
        /// </summary>
        Created,
        Dealing,
        /// <summary>
        /// 已执行,但尚未完成
        /// 下次还会继续执行
        /// </summary>
        Dealed,
        /// <summary>
        /// 已经执行，并达成目标状态
        /// </summary>
        Done,
        /// <summary>
        /// 中途中止了
        /// </summary>
        Canceled,
        Error
    }
}

File: java/workflow/src/main/java/yitek/workflow/core/ActivityStates.java
package yitek.workflow.core;

public enum ActivityStates {
	error(-1),
	created(0),
	entried(1),
	dealing(2),
	dealed(3),
	exited(4);
	private int value;
	private ActivityStates(int value) {     //必须是private的，否则编译错误  
        this.value = value;  
    }  
  
    public static ActivityStates valueOf(int value) {   //手写的从int到enum的转换函数  
        switch (value) {  
			case -1: return error;
			case 0 : return created;
        	case 1: return entried;  
        	case 2:  return dealing;
			case 3: return dealed;
			case 4: return exited;  
        default:  
            return null;  
        }  
    }  
  
    public int value() {  
        return this.value;  
    } 
}
